src/Snake.DataAccess/IUnitOfWork.cs

```diff
@@ -0,0 +1,12 @@
+﻿using Snake.DataAccess.Repositories;
+using Snake.Game.Models;
+
+namespace Snake.DataAccess
+{
+    public interface IUnitOfWork
+    {
+        GenericRepository<User> UserRepository { get; }
```

Comment:
Make this property of interface type, not class. Moreover, implement a generic method for getting repository by entity type, like:
``IRepository<TEntity> GetReository<TEntity>()``